kong/client.go

```diff
@@ -87,7 +92,16 @@ type Status struct {
 // NewClient returns a Client which talks to Admin API of Kong
 func NewClient(baseURL *string, client *http.Client) (*Client, error) {
```

Comment:
I second Travis's point about making it possible to adjust the timeout as the client, and recommend the solution below.

Expecting the user to adjust the timeout in a returned client isn't perfect either because it's at odds with the open-closed principle.

What I'd recommend would be to provide an "overloaded" `NewClient`:

```go
func NewClient(...) (*Client, error) {
  return NewClientWithTimeout(defaultTimeout, ...)
}

func NewClientWithTimeout(timeout time.Duration, ...) (*Client, error) {
  // actual implementation goes here
}
```

Reply:
This seems reasonable at first glance but on a closer inspection, it seems an unnecessary addition to the API surface.
An `*http.Client` has numerous knobs for tweaking network behavior and addition a method for each to this API seems unnecessary burden.
The reason we accept an `*http.Client` in the constructor is so that the user of the API can define all client-specific behavior as they see fit. This is how this client is used in all downstream repositories I know.
We may add `WithTimeout` function today but then that opens the door for WithXs for future and adding those here doesn't really make things better for us or the end user in my view.

For reference, here is another API that does this the same way: https://pkg.go.dev/github.com/google/go-github/v35/github#NewClient

(go-kong is inspired heavily from that API)